win32ss/user/winsrv/usersrv/shutdown.c

```diff
@@ -811,20 +811,43 @@ CSR_API(SrvExitWindowsEx)
     NTSTATUS Status;
     PUSER_EXIT_REACTOS ExitReactOSRequest = &((PUSER_API_MESSAGE)ApiMessage)->Data.ExitReactOSRequest;
 
+    Status = NtUserSetInformationThread(NtCurrentThread(),
```

Comment:
This call may be needed instead in UserClientShutdown() because it's at this point that dialog boxes about killing unresponding apps will be shown (and in principle this should switch desktops to where these apps are, and display the dialog there).